### DISCUSSION TITLE
Poetry Error for pybricks-micropython repository

### DISCUSSION THREAD

**Discussion by Redstonetill** [pybricks/support#1776] [Q&A] [answered]
I am trying to build my own version of the firmware for my spike prime hub. For that I have followed the 'CONTRIBUTING.md' in the repository. But when I am trying to setup the environment I am getting the following error:
/pybricks-micropython$ poetry env info

  RuntimeError

  The Poetry configuration is invalid:
    - Additional properties are not allowed ('package-mode', 'group' were unexpected)

  at /usr/lib/python3/dist-packages/poetry/core/factory.py:43 in create_poetry
       39│             message = ""
       40│             for error in check_result["errors"]:
       41│                 message += "  - {}\n".format(error)
       42│
    →  43│             raise RuntimeError("The Poetry configuration is invalid:\n" + message)
       44│
       45│         # Load package
       46│         name = local_config["name"]
       47│         version = local_config["version"]
       
I am using Ubuntu 22.04 (Jammy Jellyfish) and have installed every requirement, just with the difference that I installed poetry through 'apt' because it didnt find poetry through the 'pipx'-installer. Just some generell informations about the system are
- Poetry version 1.1.12
- Python 3.10.12 (is being called with 'python3')
- Uncrustify-0.72.0_f

Systeminfo:
- 4gb of Ram
- 32gb Disksize
- 1 Core

=> Ubuntu 22.04 runs on a proxmox VM (so it is a brand new installation but "apt update & apt upgrade" have been run)

Thanks in advance if anyone can help :)

**Comment by dlech** [comment, marked as answer]
It looks like you are using an old version of Poetry. Try v1.8x.

**Reply by Redstonetill** [reply]
Yes this was the problem thanks. I fixed it (just in case if anyone finds themself in the same situation :) by "apt remove python3-poetry" and then running "pipx install poetry==1.8.0 --force" and to add it to path (Which was the problem why it couldn't be found through 'pipx') I run "pipx ensurepath" and restarted my ssh session. After this I could follow the tutorial without any problems so far
